controllers/keda/scaledobject_controller.go

```diff
@@ -637,3 +632,10 @@ func (r *ScaledObjectReconciler) updateStatusWithTriggersAndAuthsTypes(ctx conte
 
 	return kedastatus.UpdateScaledObjectStatus(ctx, r.Client, logger, scaledObject, status)
 }
+
+func (r *ScaledObjectReconciler) getFromCacheOrDirect(ctx context.Context, key client.ObjectKey, obj client.Object) error {
+	// r.Client from controller-runtime here had implemented the logic to choose to perform a cache or live lookup.
+	// this functionality is mostly used in the admission webhooks or places where we actually want to implement this on our own
+	err := r.Client.Get(ctx, key, obj, &client.GetOptions{})
+	return err
```

Comment:
this function uses only the cached client, right?

The name `getFromCacheOrDirect` would suggest it uses the cached client first and a direct client as a backup when `NotFound` is returned from the cached client similar to what KEDA has implemented for the webhooks but this doesn't seem to be the case here.